core/src/impl/Kokkos_Atomic_Load.hpp

```diff
@@ -96,7 +96,7 @@ KOKKOS_INTERNAL_INLINE_DEVICE_IF_CUDA_ARCH T _atomic_load(
 #ifndef KOKKOS_IMPL_YOLO_ASSUME_TRIVIALLY_COPYABLE_TO_WORK_AROUND_BUG
               std::is_trivially_copyable<T>::value
 #else
-              true
+              false
```

Comment:
How was this not an issue before? `true` was effectively the behavior before the fixes were put in...

This is clearly what we should be doing, but I was going for minimal changes to existing behavior.

Reply:
THe problem is that `__atomic_load` will check for is_trivially_copyable (which is super ironic, because it complains about not being is_trivially_copyable, while using is_trivially_copyable as a check will not compile since it doesn't exist in the header. They must be using some other mechanism for the test inside `__atomic_load`